### PR TITLE
Move methods from Symbols/SymbolVisitors to Symbol

### PR DESCRIPTION
**Pull request by mfussenegger**
It was a bit unclear if something belongs to either Symbol, Symbols,
SymbolVisitors

This attempts to clarify it:

- Symbol: anything operating on a symbol node (either tree, or leaf)
- Symbols: anything operating on a collection of symbols (e.g. List of symbols)
- SymbolVisitors: going to be removed (this PR moves parts, but not all yet)

### CODE REVIEW

File: server/src/main/java/io/crate/expression/symbol/Symbols.java [hunk 1]
```diff
@@ -137,6 +102,19 @@ public static void toStream(Collection<? extends Symbol> symbols, StreamOutput o
         }
     }
 
+    public static List<Symbol> listFromStream(StreamInput in) throws IOException {
+        return in.readList(Symbols::fromStream);
+    }
+
+    @Nullable
+    public static Symbol nullableFromStream(StreamInput in) throws IOException {
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/expression/symbol/Symbols.java:110]: Single value streamer functions are still here - should I move these too? Would kinda have to going by the PR descriptions split decision

File: server/src/main/java/io/crate/expression/symbol/WindowFunction.java [hunk 2]
```diff
@@ -80,6 +81,36 @@ public Boolean ignoreNulls() {
         return ignoreNulls;
     }
 
+    @Override
+    public boolean any(Predicate<? super Symbol> predicate) {
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/expression/symbol/WindowFunction.java:85]: What's not so nice about this change is that these `any` implementations now duplicate logic in `DefaultTraversalSymbolVisitor`

I suspect we could either implement `any` in terms of a `DefaultTraversalSymbolVisitor`, or we could go the other way around, and try to implement more operations in terms of `any`. Given that it visits each node in a tree that should work - and it even has short-circuit capabilities. E.g. to replace `RefVisitor` we could have:

```java
    default void visitRefs(Consumer<? super Reference> consumer) {
        any(s -> {
            if (s instanceof Reference ref) {
                consumer.accept(ref);
            }
            return false;
        });
    }
```

File: server/src/main/java/io/crate/expression/symbol/Symbols.java (hunk 1 above)
Comment (seut) [reply server/src/main/java/io/crate/expression/symbol/Symbols.java:110]: You mean to move them to `Symbol`?

**Review by seut — APPROVED** [review]
nice

File: server/src/main/java/io/crate/expression/symbol/Symbols.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/expression/symbol/Symbols.java:110]: yes

Comment (seut) [reply server/src/main/java/io/crate/expression/symbol/Symbols.java:110]: Yep makes sense to do.

**Pull request merged**